core/src/main/java/com/linecorp/armeria/client/endpoint/AbstractEndpointSelector.java

```diff
@@ -26,21 +28,31 @@
 import java.util.function.Consumer;
 
 import com.google.common.annotations.VisibleForTesting;
+import com.google.errorprone.annotations.concurrent.GuardedBy;
 
 import com.linecorp.armeria.client.ClientRequestContext;
 import com.linecorp.armeria.client.Endpoint;
 import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.common.annotation.UnstableApi;
 import com.linecorp.armeria.common.util.UnmodifiableFuture;
 import com.linecorp.armeria.internal.client.ClientPendingThrowableUtil;
+import com.linecorp.armeria.internal.common.util.IdentityHashStrategy;
+import com.linecorp.armeria.internal.common.util.ReentrantShortLock;
+
+import it.unimi.dsi.fastutil.objects.ObjectLinkedOpenCustomHashSet;
 
 /**
  * A skeletal {@link EndpointSelector} implementation. This abstract class implements the
  * {@link #select(ClientRequestContext, ScheduledExecutorService)} method by listening to
  * the change events emitted by {@link EndpointGroup} specified at construction time.
  */
-public abstract class AbstractEndpointSelector implements EndpointSelector {
+public abstract class AbstractEndpointSelector implements EndpointSelector, Consumer<List<Endpoint>> {
```

Comment:
Could you define a listener field instead of implementing it? i.e.

```java
public abstract class AbstractEndpointSelector implements EndpointSelector {
    private final Consumer<List<Endpoint>> listener = endpoints -> {
        updateNewEndpoints(endpoints);
        ...
    }
}
```

Reply:
Sounds good. Method reference is used as a listener instead.